### ISSUE TITLE
App dies in [self swizzled_addSubview:subview];

### ISSUE THREAD

**Issue by ghost** [danielamitay/DAKeyboardControl#25]
Hi there,

I was trying to use DAKeyboardControl, but my App crashes after the second load of DAKeyboardControl.
To show what I mean, I modified the Example project. You can find it here:
https://github.com/appcominteractive/DAKeyboardControl/tree/swizzeledError

To see the crash: push the new push button, go back, and press the push button again. 
Tested with Xcode 4.5.2 on iPhone 5.0 and iPhone 6.0 Simulator.

Regards,
Christian

**Comment by Ashleyjwsi** [comment]
Hi
My app is crashing for the same issue. I have write [self.view removeKeyboardControl]
but still getting crash.
Kindly help.

**Comment by joshuafeldman** [comment]
Finally figured this one out, and it was really difficult. It seems like some of the views in the Apple keyboard window don't actually respond to `UIResponder` properties and methods (no idea how this is possible, Apple does some crazy internal stuff). Pretty sure it is an Apple bug.

What we did to work around the bug is create a category on `UIResponder` and swizzled the properties / methods that get accessed and wrapped the call to the original method inside of a `respondsToSelector:` and it seems to have worked nicely.

**Comment by mfarhand** [comment]
@joshuafeldman  i can't understand completely  your last comment. 
can you make  conclusion for this issue  ? 
Finally can you give us a sample about you solution category on UIResponder  ? 
but i think my problem a little different 
this is my stack trace in Fabric. this method have called over 500 time !!!
http://uupload.ir/files/56g_screen_shot_2015-10-03_at_3.25.04_pm.png

**Comment by mfarhand** [comment]
but i think my problem a little different 
this is my stack trace in Fabric
http://uupload.ir/files/56g_screen_shot_2015-10-03_at_3.25.04_pm.png

**Comment by mfarhand** [comment]
is there  any respond to my question ?

**Comment by joshuafeldman** [comment]
@mfarhand we ended up removing `DAKeyboardControl` since iOS 7+ supports the pan gesture to dismiss the keyboard without requiring this library.

But while we had it implemented we used something like this to prevent that bug where a weird subclass of `UIView` actually doesn't respond to the `isFirstResponder` method.

`-swizzleInstanceMethod:toMethod:` is a category method on `NSObject` that I didn't include in the snippet below.

``` objective-c
@implementation UIResponder (KeyboardViewResponderFix)

+ (void)load
{
    SEL originalSelector = @selector(isFirstResponder);
    SEL swizzledSelector = @selector(swizzled_isFirstResponder);

    [self swizzleInstanceMethod:swizzledSelector toMethod:originalSelector];
}

- (BOOL)swizzled_isFirstResponder
{
    if ( [self respondsToSelector:@selector(swizzled_isFirstResponder)] ) {
        return [self swizzled_isFirstResponder];
    }

    return NO;
}

@end
```

**Comment by mfarhand** [comment]
@joshuafeldman  thanks for your reply 
i have 2 question :
1) i can't understand your mean about :
-swizzleInstanceMethod:toMethod: is a category method on NSObject that I didn't include in the snippet below.
is it enough if create a category from NSObject  & swizling like your code or i need extra code  ?

what is the best solution for handle without this library?(we we to set out table view on top of keyboard frame & dismiss it with panning gesture

**Comment by joshuafeldman** [comment]
@mfarhand yes if you have your own method to swizzle just use it there. If your applications supports iOS 7+ I highly suggest looking at https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIScrollView_Class/#//apple_ref/occ/instp/UIScrollView/keyboardDismissMode

**Comment by mfarhand** [comment]
@joshuafeldman tnx for your help , i will try it  and notify you about this issue.
